lib/dataset/bins_reduction.cpp

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Jan-Lukas Wynen
+#include "scipp/dataset/bins_reduction.h"
+
+#include "scipp/variable/bins.h"
+#include "scipp/variable/creation.h"
+#include "scipp/variable/reduction.h"
+
+#include "../variable/operations_common.h"
+#include "scipp/dataset/data_array.h"
+#include "scipp/dataset/dataset.h"
+
+namespace scipp::variable {
+
+namespace {
+Variable reduce_bins(const Variable &data,
+                     void (&op)(Variable &, const Variable &),
+                     const FillValue init) {
+  auto reduced = make_reduction_accumulant(data, data.dims(), init);
+  reduce_into(reduced, data, op);
+  return reduced;
+}
+} // namespace
+
+Variable bins_sum(const Variable &data) {
+  return reduce_bins(data, variable::sum_into, FillValue::ZeroNotBool);
+}
+
+Variable bins_max(const Variable &data) {
+  return reduce_bins(data, variable::max_into, FillValue::Lowest);
```

Comment:
The operation (here `max_into`) and the fill value are always tied, and yet we have to pass them separately. Maybe we should refactor to a safer approach that cannot be misused?

Reply:
Turn `max_into` into an object with an `init` or `make_accumulant` member? I don't see how else we would tie the two together.

Reply:
Leave it, it is probably fine.